lib/Service.php

```diff
@@ -189,6 +189,30 @@ function write($rootElementName, $value, $contextUri = null) {
 
     }
 
+    /**
+     * Maps the given value-object to the given rootElementName within the given namespace.
+     *
+     * The $rootElement must be specified in clark notation.
+     *
+     * The $className refernces a regular php class which must provide a __construct() without arguments.
+     * This methods creates instances of this class which later on picks up all xml child elements as public properties.
+     *
+     * @param string $rootElementName
+     * @param object $className
+     * @param string $namespace
+     */
+    function mapValueObject($rootElementName, $className, $namespace) {
+        if (!class_exists($className)) {
+            throw new \InvalidArgumentException('class "' . $className . '" does not exist');
+        }
+        $this->namespaceMap[$namespace] = null;
```

Comment:
Well I think I would just want to encourage people to always (manually) set their preferred namespace map.

I would also suggest to always use some prefix, and not default to no prefix at all. The fallback with aliases is specifically added because it generates valid (but somewhat ugly) xml, but you, as the xml author, need to pick your preferred namespace prefix (even if that prefix is empty).

So yea this is definitely by design. We